core/sdk/src/utils/test_utils.rs

```diff
@@ -29,3 +30,26 @@ where
         panic!("Test failed for ExtnRequest variant: {:?}", request);
     }
 }
+
+pub fn log_memory_usage(label: &str) {
```

Comment:
We don't want to use this utility in production code. Please move this under a feature flag and enable that flag only for performance test purpose. 

CC @brendanobra